ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RequestHandler.java

```diff
@@ -76,14 +83,22 @@ public RequestHandler(
 
   public KsqlEntityList execute(
       final List<ParsedStatement> statements,
-      final Map<String, Object> propertyOverrides
+      final Map<String, Object> overrides
+  ) {
+    return execute(statements, overrides, injectorFactory.apply(ksqlEngine, serviceContext));
+  }
+
+  private KsqlEntityList execute(
+      final List<ParsedStatement> statements,
+      final Map<String, Object> propertyOverrides,
+      final Injector injector
   ) {
     final Map<String, Object> scopedPropertyOverrides = new HashMap<>(propertyOverrides);
```

Comment:
We could change this to be more functional now, where we pull the new configs out of the returned `ConfiguredStatement` after each call to inject(), and use those to create the next ConfiguredStatement (rather than having inject() modify the properties as a side-effect).

Reply:
We need to persist the set/unset properties across requests as well so I'm not sure how this would work

Reply:
Oh ignore that previous comment - it persists it at the CLI level, not the server. I will update this :)

Reply:
One more time... this gets a little tricky with the run script functionality since we want runscript and multiline statements to work with one another. I'll hold off on this change for another time.

Reply:
Why's that? Doesn't run-script just parse out all the statements and then call this execute method?

Reply:
The way the code is structured is:
```java
List<Entitiy> execute(List<Statements>):
   result
   for each statement:
     statement = inject(statement)
      if statement is runscript: result.add(executeRunScript(statement).last)
      else result.add(executeOneStatement(statement))
      // update configs for next statement based on current statement

List<Entity> executeRunScript(statement):
   return execute(extractRunScript(statement))
```
I would need to restructure the code so that executeRunScript returns both a list of entities and the final configured statement if I wanted to add the commented out line. It's possible, but requires some code jitter